examples/protyping/auto-segmentation/dicom_process.py

```diff
@@ -0,0 +1,383 @@
+def get_pixel_array(dicom_series):
+    """
+    Return pixel array volume from DICOM imaging series.
+    """
+    return np.array([dicom.pixel_array for dicom in dicom_series])
+
+
+def read_structures(dicom_structures):
+    structures = []
+    """Returns a sorted list of all files names in data_path with (optional) ext.
+
+    Parameters
+    ----------
+    dicom_structures : RS DICOM file
+
+    Returns
+    -------
+    structures : list
+        List containing a dict for each structure type found in dicom_structure
+        Each dict has keys: 'number', 'name', 'contour_points', 'color'
+
+    Note
+    -------
+    contour_points need to be transformed to pixel space for model
+    """
+    # loop through each structure type
+    for structure_index in range(len(dicom_structures.ROIContourSequence)):
+
+        structure_dict = {}
+        structure_dict["number"] = dicom_structures.ROIContourSequence[
+            structure_index
+        ].ReferencedROINumber
+        # Double check number is correct
+        assert (
+            structure_dict["number"]
+            == dicom_structures.StructureSetROISequence[structure_index].ROINumber
+        )
+        structure_dict["name"] = dicom_structures.StructureSetROISequence[
+            structure_index
+        ].ROIName
+        structure_dict["contour_points"] = [
+            z_slice.ContourData
+            for z_slice in dicom_structures.ROIContourSequence[
+                structure_index
+            ].ContourSequence
+        ]
+        structure_dict["color"] = dicom_structures.ROIContourSequence[
+            structure_index
+        ].ROIDisplayColor
+
+        structures.append(structure_dict)
+    return structures
+
+
+def transform_to_array(x, y, dicom_series):
+    """
+    Transform from patient space to pixel space
+    """
+    translation = dicom_series[0].ImagePositionPatient
+    scale = dicom_series[0].PixelSpacing
+    orientation = dicom_series[0].ImageOrientationPatient
+    x = np.array(x)
+    y = np.array(y)
+
+    # NOTE Only handles +-1 cosines
+    # A more robust method that handles intermediate angels
+    # was attempted however the affine matrix was singular
+    # See: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.2.htmlx1
+    # See: https://dicomiseasy.blogspot.com/2013/06/getting-oriented-using-image-plane.html
+    r = (y - translation[1]) / scale[1] * orientation[4]
+    c = (x - translation[0]) / scale[0] * orientation[0]
+
+    return r, c
+
+
+def get_binary_masks(structures, dicom_series, images):
+    # Should z location for each slice in patient space
+    z_slice_locations = [(dicom.ImagePositionPatient[2]) for dicom in dicom_series]
+    # an array full of zeros the same shape as images
+    labels = np.zeros_like(images, dtype=np.int16)
+
+    # NOTE: This function only works for a single label volume output
+    # See line for structure in structures[0:1] to choose
+    for structure in structures[0:1]:
```

Comment:
Think the unix philosophy of "do one job, and do it well" for functions too. Ideally one could read the name of a function and pretty much know everything it should run off and do. Anything that doesn't come to mind, it shouldn't really go and do in secret. If a function accepts a list of structures, as the caller of the API I would expect it to return to me a list of results on all those structures (or raise an error if it could not). If the function is designed to only return one structure, then have the API only receive one structure also :).